### ISSUE TITLE
Dio not working with web

### ISSUE THREAD

**Issue by nisa10880** [flutterchina/dio#1164]
I tried to make dio work with web but I had this exception

`Exception has occurred. "Error: Unsupported operation: Platform._version at Object.throw_ [as throw] (http://localhost:51148/dart_sdk.js:5037:11) at Function._version (http://localhost:51148/dart_sdk.js:53428:17) at Function.get version [as version] (http://localhost:51148/dart_sdk.js:53502:27) at get _version (http://localhost:51148/dart_sdk.js:53372:27) at Function.desc.get [as _version] (http://localhost:51148/dart_sdk.js:5526:17) at Function.get version [as version] (http://localhost:51148/dart_sdk.js:53346:26) at Object._getHttpVersion (http://localhost:51148/dart_sdk.js:188568:31) at new _http._HttpClient.new (http://localhost:51148/dart_sdk.js:183769:28) at Function.new (http://localhost:51148/dart_sdk.js:178274:16) at io_adapter.DefaultHttpClientAdapter.new.[_configHttpClient] (http://localhost:51148/packages/dio/src/adapters/io_adapter.dart.lib.js:169:56) at io_adapter.DefaultHttpClientAdapter.new.fetch (http://localhost:51148/packages/dio/src/adapters/io_adapter.dart.lib.js:64:50) at fetch.next (<anonymous>) at runBody (http://localhost:51148/dart_sdk.js:37393:34) at Object._async [as async] (http://localhost:51148/dart_sdk.js:37424:7) at io_adapter.DefaultHttpClientAdapter.new.fetch (http://localhost:51148/packages/dio/src/adapters/io_adapter.dart.lib.js:60:20) at dio_service.AppDio.__._dispatchRequest (http://localhost:51148/packages/dio/src/interceptors/log.dart.lib.js:1254:56) at _dispatchRequest.next (<anonymous>) at http://localhost:51148/dart_sdk.js:37374:33 at _RootZone.runUnary (http://localhost:51148/dart_sdk.js:37245:59) at _FutureListener.thenAwait.handleValue (http://localhost:51148/dart_sdk.js:32501:29) at handleValueCallback (http://localhost:51148/dart_sdk.js:33028:49) at Function._propagateToListeners (http://localhost:51148/dart_sdk.js:33066:17) at _Future.new.[_completeWithValue] (http://localhost:51148/dart_sdk.js:32914:23) at async._AsyncCallbackEntry.new.callback (http://localhost:51148/dart_sdk.js:32935:35) at Object._microtaskLoop (http://localhost:51148/dart_sdk.js:37497:13) at _startMicrotaskLoop (http://localhost:51148/dart_sdk.js:37503:13) at http://localhost:51148/dart_sdk.js:33274:9
`

### New Issue Checklist

- [x] I have searched for a similar issue in the [project](https://github.com/flutterchina/dio/issues) and found none

### Issue Info

| Info                           | Value                                                 |      |
| ------------------------------ | ----------------------------------------------------- | ---- |
| Platform Name                  | e.g. flutter / ios / android                          | web |
| Platform Version               | e.g.  1.5.0 / 12.0 / 9.0                              |      |
| Dio Version                    | e.g.  2.1.0 / 1.0.17                                  |    4.0.0  |
| Android Studio / Xcode Version | e.g. Android Studio 3.3.2 / Xcode 10.2.1              |      |
| Repro rate                     | e.g. all the time (100%) / sometimes x% / only once   |      |
| Repro with our demo prj        | e.g. does it happen with our demo project?            |      |
| Demo project link              | e.g. link to a demo project that highlights the issue |      |

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

**Comment by kateile** [comment]
that's not dio. its seems that you are using `Platform` on web code. calling something like `if(Platform.isAndroid){}` will result into the same error. so remove Platform from your code and use something like `kIsWeb` or post minimun dio code and I will tell you where error comes from

**Comment by nisa10880** [comment]
Thank you. I removed all the platform.android part of my code but the error is not from there. The problem is from my Dio configuration, I still don't understand why my configuration works only for IOS/android. 

Here is the implemention.

`final dioProvider = Provider<Dio>((_) => AppDio.getInstance());

class AppDio with DioMixin implements Dio {
  Future<String?> _getToken() async {
    final prefs = await SharedPreferences.getInstance();
    final token = prefs.get('token');
    return token as Future<String?>;
  }

  AppDio._([BaseOptions? options]) {
    options = BaseOptions(
      baseUrl: apiUrl,
      contentType: 'application/json',
      connectTimeout: 20000,
      sendTimeout: 20000,
      receiveTimeout: 20000,
    );

    this.options = options;

    this.httpClientAdapter = DefaultHttpClientAdapter();
  }

  static Dio getInstance() {
    return AppDio._()
      ..interceptors.add(
        InterceptorsWrapper(
          onRequest: (options, handler) async {
            if (options.headers.containsKey("requiresToken")) {
              //remove the auxiliary header
              options.headers.remove("requiresToken");

              final SharedPreferences prefs =
                  await SharedPreferences.getInstance();
              final token = prefs.get('token');

              options.headers.addAll({"Authorization": "Bearer $token"});

              return handler.next(options);
            }
            handler.next(options);
          },
        ),
      );
    // ..interceptors.add(
    //     DioCacheManager(CacheConfig(baseUrl: Constants.of().endpoint))
    //         .interceptor as Interceptor)
    //  ..interceptors.add(LogInterceptor(responseBody: true));
  }
}
`

**Comment by SYThomasNordmeyer** [comment]
Was this solved? I am getting the same error message.

**Comment by SYThomasNordmeyer** [comment]
OK, just saw, I am using DioForNative, as I re-used some code from App development. I guess this just doesn't work for web.
